module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -117,126 +118,155 @@ namespace module::vision {
                         deviation += (mean - angle) * (mean - angle);
                     }
                     deviation = std::sqrt(deviation / (angles.size() - 1));
 
+                    // If the deviation is more than the maximum allowed, then discard this ball
                     if (deviation > config.maximum_deviation) {
 
                         log<NUClear::DEBUG>(fmt::format("Ball discarded: deviation ({}) > maximum_deviation ({})",
                                                         deviation,
                                                         config.maximum_deviation));
                         log<NUClear::DEBUG>("--------------------------------------------------");
+                        // Balls that violate degree of fit will show as green in NUsight
                         b.colour = keep ? message::conversion::math::fvec4(0.0f, 1.0f, 0.0f, 1.0f) : b.colour;
                         keep     = false;
                     }
 
-                    // DISTANCE IS TOO CLOSE
+                    // DISCARD IF THE DISTANCE FROM THE BALL TO THE ROBOT IS TOO CLOSE
+                    // Prevents the robot itself being classed as a ball, ie its arms/hands
                     if (distance < config.minimum_ball_distance) {
 
                         log<NUClear::DEBUG>(fmt::format("Ball discarded: distance ({}) < minimum_ball_distance ({})",
                                                         distance,
                                                         config.minimum_ball_distance));
                         log<NUClear::DEBUG>("--------------------------------------------------");
+                        // Balls that violate this but not previous checks will show as red in NUsight
                         b.colour = keep ? message::conversion::math::fvec4(1.0f, 0.0f, 0.0f, 1.0f) : b.colour;
                         keep     = false;
                     }
 
-                    // IF THE DISAGREEMENT BETWEEN THE ANGULAR AND PROJECTION BASED DISTANCES ARE TOO LARGE
-                    // Intersect cone axis vector with a plane midway through the ball with normal vector (0, 0,
+                    // DISCARD IF THE DISAGREEMENT BETWEEN THE ANGULAR AND PROJECTION BASED DISTANCES ARE TOO LARGE
+                    // Intersect cone axis vector with a plane midway through the ball with normal vector (0, 0, 1)
                     // 1) Do this in world space, not camera space!
                     // https://en.wikipedia.org/wiki/Line%E2%80%93plane_intersection#Algebraic_form
                     // Plane normal = (0, 0, 1)
                     // Point in plane = (0, 0, field.ball_radius)
                     // Line direction = axis
-                    // Point on line = camera = Hcw.topRightCorner<3, 1>()
+                    // Point on line = camera = Hwc.translation.z() = rCWw.z()
                     Eigen::Affine3f Hcw(horizon.Hcw.cast<float>());
```

Comment:
This has been used a few times, maybe just make an Affine3f right up the top of the loop once and reuse it